### PR TITLE
Untangle the linkToDocs method in OC_Helper

### PR DESCRIPTION
**Pull request by MorrisJobke**
- now uses the proper URLGenerator interface

cc @rullzer @LukasReschke @nickvergessen @PVince81

### CODE REVIEW

File: lib/private/config.php [hunk 1]
```diff
@@ -234,7 +234,7 @@ private function writeData() {
 
 		// File does not exist, this can happen when doing a fresh install
 		if(!is_resource ($filePointer)) {
-			$url = \OC_Helper::linkToDocs('admin-dir_permissions');
+			$url = \OC::$server->getURLGenerator()->linkToDocs('admin-dir_permissions');
```
Comment (rullzer) [inline lib/private/config.php]: Can't we also inject the urlGenerator?

**Comment by rullzer** [comment]
:+1: Nice cleanup

File: lib/private/config.php (hunk 1 above)
Comment (nickvergessen) [inline lib/private/config.php]: no, no, no :speak_no_evil:

**Comment by nickvergessen** [comment]
What exactly was the problem we had last time with the doc links?

**Comment by PVince81** [comment]
> What exactly was the problem we had last time with the doc links?

Total destruction of routing because the router isn't initialized at this time.
CC @Xenopathic

**Comment by MorrisJobke** [comment]
> Total destruction of routing because the router isn't initialized at this time.

Which shouldn't be the case here, because I only moved the call one level up (from a nested method call to the actual place where the wrapper method is called)

File: lib/private/config.php (hunk 1 above)
Comment (MorrisJobke) [inline lib/private/config.php]: I guess this can lead to weird problems of initialization. I would like to keep it as it is now.

**Comment by nickvergessen** [comment]
> Which shouldn't be the case here, because I only moved the call one level up (from a nested method call to the actual place where the wrapper method is called)

Right, but that's why DI won't work. I think we should add that as a comment, so no one tries to move it and thereby actually breaks all the apps

**Comment by MorrisJobke** [comment]
> Right, but that's why DI won't work. I think we should add that as a comment, so no one tries to move it and thereby actually breaks all the apps

It will work once everything is setup through DI...but during this state of mixed code it won't work. I will add a comment

**Comment by MorrisJobke** [comment]
@nickvergessen Mind to review?

**Comment by MorrisJobke** [comment]
@nickvergessen I added the suggested comment regarding DI problems

**Comment by nickvergessen** [comment]
:+1:

**Pull request merged**